src/main/scala/com/modelfabric/sparql/api/SparqlQuery.scala

```diff
@@ -34,6 +35,7 @@ object SparqlQuery {
     * @param query
     * @return
     */
+  // JC: if we define SparqlQuery as a case class, then we don't need to define this unapply method
```

Comment:
The idea was to create subclasses of SparqlQuery, which would use their own bespoke solution mappers, i.e you could have  a CustomerQuery that returned Customer objects